### PR TITLE
Adding instrucitons for setting up AzureCNI private cluster.

### PR DESCRIPTION
**Pull request by clomeli-ms**
Adds instructions setting up AzureCNI private cluster with a private link workspace some customers requested.

### CODE REVIEW

File: docs/azure-cni-setup.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+The VNET does need to have a sufficiently  large address space, The default address space of 10.0.0.0/16 has worked in our testing. The subnet also needs to be large enough to accommodate the AKS cluster, In general, you'll want 128 addresses per node in your cluster, if using default settings. More documentation on creating an Azure VNET can be found here:
+https://docs.microsoft.com/en-us/azure/virtual-network/quick-create-portal.
+
+### Setting up AKS cluster:
+
+Setting up your AKS cluster with Azure CNI is documented here:
+https://docs.microsoft.com/en-us/azure/aks/configure-azure-cni
+
+When creating the cluster make sure to check the "Enable private cluster" checkbox under the Networking section in the Azure Portal. As mentioned in the VNET setup section of this document the VNET being used can either be created independently or during setup of the AKS cluster through the portal. The portal has the added convenience of making sure your subnet is large enough for your cluster.
```
Comment (mjaow) [inline docs/azure-cni-setup.md:17]: I think currently we're targeting to AKS with azureCNI ,right? As cni aks has public ip and we did verify it, but for private aks, it doesn't has public ip and we haven't verified yet

File: docs/azure-cni-setup.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+### Installing AMLArc:
+
+All instructions for connecting to an AKS AzureCNI cluster are the same as a public AKS cluster:
+https://docs.microsoft.com/en-us/azure/machine-learning/how-to-attach-arc-kubernetes
+
+Do take note however that all commands must be run from a VM within the VNET to be able to communicate and install the Arc and AMLArc extensions on the cluster.
+
+### Setting up AzureML Workspace:
+
+Private link workspace which is required if using an AzureCNI cluster.
```
Comment (mjaow) [inline docs/azure-cni-setup.md:28]: 1. private link workspace is required only if user want to secure their connection to workspace
2. user can use cni aks ,but also a public workspace

File: docs/azure-cni-setup.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+Documentation for a private link workspace is provided here:
+https://docs.microsoft.com/en-us/azure/machine-learning/how-to-configure-private-link?tabs=python
+
+Notably, if one wants to access the workspace on a browser one has to either enable public access to the cluster, or connect to the VNET with. How to enable public access is described in the document above. Notably if using the Python SDK the Workspace.update() command should be called from a VM or other machine in the VNET.
```
Comment (mjaow) [inline docs/azure-cni-setup.md:32]: I think what you mean is access from studio,right? here is a related doc for the notice https://docs.microsoft.com/en-us/azure/machine-learning/how-to-enable-studio-virtual-network

File: docs/azure-cni-setup.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# AMLArc with AzureCNI AKS cluster and Private Link AML Workspace
```
Comment (mjaow) [inline docs/azure-cni-setup.md:1]: it seems here is a guide of how to secure workspace/aks, do we also need a full-guide for others workspace-related resources like blobstorage/acr/keyvault

File: docs/azure-cni-setup.md (hunk 3 above)
Comment (clomeli-ms) [reply docs/azure-cni-setup.md:32]: Yes, I'll add that.

File: docs/azure-cni-setup.md (hunk 1 above)
Comment (clomeli-ms) [reply docs/azure-cni-setup.md:17]: In my testing it only worked with private aks, perhaps i made a mistake when testing standard aks. Were you able to successfully install extension in that scenario?

File: docs/azure-cni-setup.md (hunk 2 above)
Comment (clomeli-ms) [reply docs/azure-cni-setup.md:28]: I'll add this clarification.

File: docs/azure-cni-setup.md (hunk 4 above)
Comment (clomeli-ms) [reply docs/azure-cni-setup.md:1]: Instructions for these other resources haven't been requested by users yet (to my knowledge), if they are we can expand the docs.

**Pull request merged**